viv_utils/emulator_drivers.py

```diff
@@ -731,6 +730,8 @@ def run(self, va: int):
 
                     # stop emulating, and go to next block in the queue.
                     break
+                except v_envi.exc.BreakpointHit:
+                    break
```

Comment:
hm, yeah, this should be handled better. though, i dont think `break` is the right thing to do. i'll study it.

Reply:
```suggestion
                except BreakpointHit as e:
                    # client indicated not to emulate beyond the given insn
                    # by raising a breakpoint (via a monitor).
                    logger.debug(
                        "driver: run_function: breakpoint: %s: 0x%x",
                        e.reason,
                        lastpc,
                    )
                    blockend = lastpc
                    for mon in self._monitors:
                        mon.postblock(self, blockstart, blockend)

                     # stop emulating, and go to next block in the queue.
                     break
```